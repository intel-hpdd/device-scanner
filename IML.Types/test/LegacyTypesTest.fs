// Copyright (c) 2018 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

module IML.Types.LegacyTypesTest

open Fable.Import.Jest
open Matchers
open Thot.Json

open IML.CommonLibrary

open LegacyTypes
open Fixtures
open IML.Types.LegacyTypes

test "decode / encode LegacyDev types" <| fun () ->
  fixtures.legacyZFSPool
    |> Decode.decodeString LegacyZFSDev.decode
    |> Result.unwrap
    |> LegacyZFSDev.encode
    // |> Encode.encode 2
    |> toMatchSnapshot
