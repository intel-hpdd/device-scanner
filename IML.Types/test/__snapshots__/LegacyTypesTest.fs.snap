// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decode / encode LegacyDev types 1`] = `
Object {
  "block_device": "zfspool:1689599553517780274",
  "drives": Array [
    "32:6",
    "32:8",
  ],
  "name": "testPool1",
  "path": "testPool1",
  "size": "20971520",
  "uuid": "1689599553517780274",
}
`;

exports[`decode / encode LegacyDev types 2`] = `
Object {
  "block_device": "zfsdataset:1689599553517780274",
  "drives": Array [
    "32:6",
    "32:8",
  ],
  "name": "testPool1/home",
  "path": "testPool1/home",
  "size": "20971520",
  "uuid": "1689599553517780274",
}
`;

exports[`decode / encode LegacyDev types 3`] = `
Object {
  "device_path": "/devices/pci0000:00/0000:00:05.0/virtio1/host2/target2:0:0/2:0:0:0/block/sda",
  "device_type": "disk",
  "dm_lv": null,
  "dm_multipath": null,
  "dm_slave_mms": Array [],
  "dm_vg": null,
  "dm_vg_size": null,
  "filesystem_type": "linux_raid_member",
  "filesystem_usage": "raid",
  "is_ro": false,
  "is_zfs_reserved": false,
  "lv_uuid": null,
  "major_minor": "8:0",
  "md_device_paths": Array [],
  "md_uuid": null,
  "parent": null,
  "partition_number": null,
  "path": "/devices/pci0000:00/0000:00:05.0/virtio1/host2/target2:0:0/2:0:0:0/block/sda",
  "paths": Array [
    "/dev/disk/by-id/scsi-0QEMU_QEMU_HARDDISK_disk1",
    "/dev/disk/by-path/virtio-pci-0000:00:05.0-scsi-0:0:0:0",
    "/dev/sda",
  ],
  "serial_80": "SQEMU    QEMU HARDDISK   disk1",
  "serial_83": "0QEMU    QEMU HARDDISK   disk1",
  "size": "20971520",
  "vg_uuid": null,
}
`;
